src/hevm/src/EVM.hs

```diff
@@ -1818,49 +1818,46 @@ cheat (inOffset, inSize) (outOffset, outSize) = do
   case fromSized <$> unliteral abi of
     Nothing -> vmError UnexpectedSymbolicArg
     Just abi ->
-              case Map.lookup abi cheatActions of
-                Nothing ->
+      case Map.lookup abi cheatActions of
+        Nothing ->
+          vmError (BadCheatCode (Just abi))
+        Just (argTypes, action) ->
+          case input of
+            SymbolicBuffer _ -> vmError UnexpectedSymbolicArg
+            ConcreteBuffer input' ->
+              case runGetOrFail
+                     (getAbiSeq (length argTypes) argTypes)
+                     (LS.fromStrict input') of
+                Right ("", _, args) -> do
+                  action outOffset outSize (toList args)
+                  next
+                  push 1
+                _ ->
                   vmError (BadCheatCode (Just abi))
-                Just (argTypes, action) ->
-                  case input of
-                    SymbolicBuffer _ -> vmError UnexpectedSymbolicArg
-                    ConcreteBuffer input' ->
-                      case runGetOrFail
-                             (getAbiSeq (length argTypes) argTypes)
-                             (LS.fromStrict input') of
-                        Right ("", _, args) ->
-                          action (toList args) >>= \case
-                            Nothing -> do
-                              next
-                              push 1
-                            Just (encodeAbiValue -> bs) -> do
-                              next
-                              modifying (state . memory)
-                                (writeMemory (ConcreteBuffer bs) outSize 0 outOffset)
```

Comment:
Do you think this was intended for return data or something else?

Reply:
Yes, I think so. n.b. that this now is now assumed to happen in the cheatAction